ua/node_id.go

```diff
@@ -118,10 +142,22 @@ func ParseNodeID(s string) (*NodeID, error) {
 	}
 
 	// parse namespace
-	var ns uint16
+	var nsid uint16
+	var nsu string
 	switch {
 	case strings.HasPrefix(nsval, "nsu="):
-		return nil, errors.Errorf("namespace urls are not supported: %s", s)
+		nsuval := strings.TrimPrefix(nsval, "nsu=")
+		ok := false
+		for id, uri := range ns {
+			if uri == nsuval {
+				nsid, nsu = uint16(id), uri
+				ok = true
+				break
+			}
+		}
+		if !ok {
+			return nil, errors.Errorf("invalid namespace uri %s", nsval)
```

Comment:
```go
return nil, errors.Errorf("namespace uri %s not found in the server NamespaceArray %s", nsval, ns)
```

Reply:
Can we use the %q instead of the %s ?

I find that avoid issues when the value is empty or when it is something that may follow the flow of the text!

Reply:
comment updated